install.sh

```diff
@@ -1,36 +1,41 @@
 #!/bin/bash
+# Check if the script is running as root
+[ "$EUID" -eq 0 ] || exec sudo bash "$0" "$@"
```

Comment:
- I'm pretty sure the os detection and the wget really don't benefit from being run as root.
- Also, some bsd machines dont have sudo by default.

It might make more sense to just wrap the `sudo` around `autorestic install`... and maybe only after checking there is a `sudo` in the first place?

Reply:
Hello, i did some changes to handle the elevation to root with sudo or su, and only the commands that needs to write to /usr/local/bin are run as root.